website/content/en/highlights/2022-05-23-0-23-0-upgrade-guide.md

```diff
@@ -203,10 +186,104 @@ And the `syslog` source returned:
 function like:
 
 ```coffeescript
 flatten(parse_syslog!(s'<1>1 2022-04-25T23:21:45.715740Z Gregorys-MacBook-Pro.local 2d4d9490-794a-4e60-814c-5597bd5b7b7d 79978 - [exampleSDID@32473 foo.baz="bar"] test message'))
 ```
+
+#### [`gcp_pubsub` sink requires setting "encoding" option] {#gcp_sink-mandatory-encoding}
+
+The `gcp_pubsub` sink now supports a variety of codecs. To encode your logs as JSON before
+publishing them to Cloud Pub/Sub, add the following encoding option
+
+```toml
+encoding.codec = "json"
+```
+
+to the config of your `gcp_pubsub` sink.
+
+#### [`humio_metrics` sink no longer has "encoding" option] {#humio_metrics-sink-fixed-encoding}
+
+The `humio_metrics` sink configuration no longer expects an "encoding" option.
+If you previously used the encoding option
+
+```toml
+encoding.codec = "json"
+```
+
+you need to remove the line from your `humio_metrics` config. Metrics are now
+always sent to Humio using the JSON format.
+
+#### [New "framing" and "encoding" options for sinks] {#sinks-framing-encoding-options}
+
+We streamlined the encoding configuration for our sinks, enabling all applicable sinks to select
+from a variety of codecs: `json`, `text`, `raw`, `logfmt`, `avro`, `native` or `native_json`, e.g.
+by setting
+
+```
+encoding.codec = "json"
+```
+
+in your sink configuration.
+
+Additionally, some sinks now support configuring how encoded events should be separated within a
+stream or batch: `bytes`, `character_delimited`, `length_delimited` or `newline_delimited`, e.g. by
+setting
+
+```
+framing.method = "newline_delimited"
+```
+
+in your sink configuration.
+
+The following sinks support setting an encoding codec: `aws_cloudwatch_logs`,
+`aws_kinesis_firehose`, `aws_kinesis_streams`, `aws_s3`, `aws_sqs`, `azure_blob`, `console`, `file`,
+`gcp_cloud_storage`, `gcp_pubsub`, `http`, `humio_logs`, `kafka`, `loki`, `nats`, `papertrail`,
+`pulsar`, `redis`, `socket`, `splunk_hec_logs` and `websocket`.
+
+Additionally, the following sinks support setting a framing method: `aws_s3`, `azure_blob`,
+`console`, `file`, `gcp_cloud_storage`, `http` and `socket`.
+
+### Deprecations
+
+#### [Shorthand values for "encoding" options deprecated] {#deprecated-encoding-shorthand}
+
+We are deprecating setting encoding options by a shorthand string. E.g. when your sink encoding used
+
+```
+encoding = "json"
+```
+
+it should now be replaced by explicitly setting the `codec`
+
+```
+encoding.codec = "json"
+```
+
+#### [Sink encoding value "ndjson" is now "json" encoding + "newline_delimited" framing] {#sink-encoding-ndjson-json}
+
+The "ndjson" encoding value will be phased out since the "newline-delimited" behavior can now be set
+in a dedicated "framing" field.
+
+Sinks that previously used
+
+```
+encoding.codec = "ndjson"
+```
+
+should now be configured as
+
+```
+framing.method = "newline_delimited"
+encoding.codec = "json"
+```
+
+if the sinks supports framing or as
+
+```
+encoding.codec = "json"
```

Comment:
Which sinks "default" to _ndjson_, can we note in their docs that json == ndjson?

Reply:
Ah, that is a good point 🤔 Will figure out the exact set of sinks that are affected and either don't have a framing option or don't need to set it explicitly

Reply:
Ah, that is a good point 🤔 Will figure out the exact set of sinks that are affected and either don't have a framing option or don't need to set it explicitly